uitools/cpp/Esri/ArcGISRuntime/Toolkit/FloorFilterController.cpp

```diff
@@ -0,0 +1,363 @@
+  {
+    return m_selectedSiteId;
+  }
+
+  void FloorFilterController::setSelectedSiteId(QString selectedSiteId)
+  {
+    if (m_selectedSiteId == selectedSiteId)
+      return;
+
+    m_selectedSiteId = std::move(selectedSiteId);
+    emit selectedSiteIdChanged();
+  }
+
+  FloorFacility* FloorFilterController::selectedFacility() const
+  {
+    if (auto floorManager = getFloorManager(m_geoView))
+    {
+      auto facilities = floorManager->facilities();
+      auto it = std::find_if(std::cbegin(facilities),
+                             std::cend(facilities),
+                             [this](FloorFacility* facility) -> bool
+                             {
+                               return facility && facility->facilityId() == selectedFacilityId();
+                             });
+      if (it != std::cend(facilities))
+      {
+        return *it;
+      }
+    }
+    return nullptr;
+  }
+
+  FloorSite* FloorFilterController::selectedSite() const
+  {
+    if (auto floorManager = getFloorManager(m_geoView))
+    {
+      auto sites = floorManager->sites();
+      auto it = std::find_if(std::cbegin(sites),
+                             std::cend(sites),
+                             [this](FloorSite* site) -> bool
+                             {
+                               return site && site->siteId() == selectedSiteId();
+                             });
+      if (it != std::cend(sites))
+      {
+        return *it;
+      }
+    }
+    return nullptr;
+  }
+
+  FloorLevel* FloorFilterController::selectedLevel() const
+  {
+    if (auto floorManager = getFloorManager(m_geoView))
+    {
+      auto levels = floorManager->levels();
+      auto it = std::find_if(std::cbegin(levels),
+                             std::cend(levels),
+                             [this](FloorLevel* level) -> bool
+                             {
+                               return level && level->levelId() == selectedLevelId();
+                             });
+      if (it != std::cend(levels))
+      {
+        return *it;
+      }
+    }
+    return nullptr;
+  }
+
+  void FloorFilterController::populateLevelsForSelectedFacility()
+  {
+    m_levels->clear();
+    auto manager = getFloorManager(m_geoView);
+    if (!manager)
+      return;
+
+    const auto allLevels = manager->levels();
+
+    QList<QObject*> levelItems;
+    for (const auto level : allLevels)
+    {
+      if (level->facility()->facilityId() == selectedFacilityId())
+      {
+        levelItems << new FloorFilterLevelItem(level, m_levels);
+      }
+    }
+    //' Reverse order for ascending order.
+    m_levels->append(QList<QObject*>(std::crbegin(levelItems), std::crend(levelItems)));
+  }
+
+  void FloorFilterController::populateFacilitiesForSelectedSite()
+  {
+    m_facilities->clear();
+    auto manager = getFloorManager(m_geoView);
+    if (!manager)
+      return;
+
+    const auto allFacilites = manager->facilities();
```

Comment:
I think it would be slightly better to find the selected site using the ID iterating over `manager->sites`, then use the list `facilities` stored into it. This comes from the fact that there would be less sites than total facilities